components/layout/construct.rs

```diff
@@ -1408,88 +1408,98 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
-                            if node.is_replaced_content() {
-                                properties::modify_style_for_replaced_content(&mut style);
+
+                        match fragment.specific {
+                            SpecificFragmentInfo::InlineBlock(ref mut inline_block_fragment) => {
+                                let flow_ref = flow_ref::deref_mut(&mut inline_block_fragment.flow_ref);
+                                flow::mut_base(flow_ref).restyle_damage.insert(damage);
+                                // FIXME(pcwalton): Fragment restyle damage too?
+                                flow_ref.repair_style_and_bubble_inline_sizes(&style);
+                            }
+                            SpecificFragmentInfo::InlineAbsoluteHypothetical(
+                                    ref mut inline_absolute_hypothetical_fragment) => {
+                                let flow_ref = flow_ref::deref_mut(
+                                    &mut inline_absolute_hypothetical_fragment.flow_ref);
+                                flow::mut_base(flow_ref).restyle_damage.insert(damage);
+                                // FIXME(pcwalton): Fragment restyle damage too?
+                                flow_ref.repair_style_and_bubble_inline_sizes(&style);
+                            }
+                            SpecificFragmentInfo::InlineAbsolute(ref mut inline_absolute_fragment) => {
+                                let flow_ref = flow_ref::deref_mut(
+                                    &mut inline_absolute_fragment.flow_ref);
+                                flow::mut_base(flow_ref).restyle_damage.insert(damage);
+                                // FIXME(pcwalton): Fragment restyle damage too?
+                                flow_ref.repair_style_and_bubble_inline_sizes(&style);
+                            }
+                            SpecificFragmentInfo::ScannedText(_) => {
+                                // Text fragments in ConstructionResult haven't been scanned yet
+                                unreachable!()
+                            }
+                            SpecificFragmentInfo::GeneratedContent(_) |
+                            SpecificFragmentInfo::UnscannedText(_) => {
+                                // We can't repair this unscanned text; we need to update the
+                                // scanned text fragments.
+                                //
+                                // TODO: Add code to find and repair the ScannedText fragments?
+                                return false
+                            }
+                            _ => {
+                                if node.is_replaced_content() {
+                                    properties::modify_style_for_replaced_content(&mut style);
+                                }
+                                fragment.repair_style(&style);
+                                set_has_newly_constructed_flow_flag = true;
```

Comment:
Why can't you just add the flag here? Why do you need to set the boolean to have it add the flag outside the match statement?